### DISCUSSION TITLE
Assigning `...` to a type statement shadows SyntaxError, is this intended?

### DISCUSSION THREAD

**Discussion by Daraan** [microsoft/pyright#9120] [Q&A] [answered]
I played around a bit and came accross the following case where for the `Y` declaration the expected warning for the SyntaxError does not show up, is this intended?

Code sample in [pyright playground](https://pyright-play.net/?pythonVersion=3.13&strict=true&code=C4TwDgpgBAGg2gFQLwEsB2wA0UECYC6USUAcgPZoQCwAUKJFAJqKobZ6HEB0PttQA)

```python
type X[T=int, T2] = None  # Wrong order, shows expected error
type Y[T=int, T2] = ...  # no error
```

**Comment by erictraut** [comment, marked as answer]
Pyright should be emitting an error here because `...` is an invalid type expression. It's internally detecting this error condition (without reporting it) and then skipping additional checks for the remainder of the `type` expression, which is why you don't see the other error that you expected.

This will be addressed in the next release.
